File: AccuPay/Repositories/PayFrequencyRepository.vb
Option Strict On

Imports System.Threading.Tasks
Imports AccuPay.Entity
Imports Microsoft.EntityFrameworkCore

Namespace Global.AccuPay.Repository

    Public Class PayFrequencyRepository

        Public Async Function GetAllAsync() As Task(Of List(Of PayFrequency))

            Using context As New PayrollContext

                Return Await context.PayFrequencies.ToListAsync

            End Using

        End Function




    End Class

End Namespace

File: Accupay.Data/Repositories/PayFrequencyRepository.cs
using AccuPay.Data.Entities;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace AccuPay.Data.Repositories
{
    public class PayFrequencyRepository
    {
        private readonly PayrollContext _context;

        public PayFrequencyRepository(PayrollContext context)
        {
            this._context = context;
        }

        public async Task<IEnumerable<PayFrequency>> GetAllAsync()
        {
            return await _context.PayFrequencies.ToListAsync();
        }
    }
}